app/routes/calendar/new.tsx

```diff
@@ -315,80 +327,91 @@ function DatesInput() {
     : "";
   const NEW_CALENDAR_EVENT_HOURS_OFFSET = 24;
 
+  const addDate = () =>
+    setDatesInputState((current) => {
+      // .reverse() is mutating, but map/filter returns a new array anyway.
+      const lastValidDate = current
+        .map((e) => e.date)
+        .filter(isDefined)
+        .reverse()
+        .at(0);
```

Comment:
https://caniuse.com/mdn-javascript_builtins_array_at

This is pretty recent addition so there is a risk of crashing I believe? Maybe we can just do [0]